diff a/483.xalancbmk/src/xercesc/util/NameIdPool.hpp b/483.xalancbmk/src/xercesc/util/NameIdPool.hpp
--- a/483.xalancbmk/src/xercesc/util/NameIdPool.hpp
+++ b/483.xalancbmk/src/xercesc/util/NameIdPool.hpp
@@ -329,7 +329,7 @@ private :
     // -----------------------------------------------------------------------
     unsigned int            fCurIndex;
     NameIdPool<TElem>*      fToEnum;
-    MemoryManager* const    fMemoryManager;
+    MemoryManager*          fMemoryManager;
 };
 
 XERCES_CPP_NAMESPACE_END